### ISSUE TITLE
UseDotNetV2 doesn't respect rollForward in global.json

### ISSUE THREAD

**Issue by vsilvar** [microsoft/azure-pipelines-tasks#17179]
**Question, Bug, or Feature?**  
*Type*: Feature

**Enter Task Name**: UseDotNet

## Environment
- Server - Azure Pipelines
- Agent - Hosted ubuntu-latest

## Issue Description

I'm using the UseDotNetV2 task with "Use Global.json" set to true. My global.json is in the solution directory and the contents are:
```
{
  "sdk": {
    "version": "6.0.403",
    "rollForward": "latestMajor"
  }
}
```
It could be argued that this should install the latest available .NET SDK (`7.0.100` at the moment), instead of `6.0.403`,
as this is the behavior of Visual Studio, and what I think a user would expect.

Another example that works in Visual Studio, but doesn't work with the current task, would be:
```
{
  "sdk": {
    "version": "7.0.0",
    "rollForward": "latestMinor"
  }
}
```
Which should be equivalent to just using the task and specifying version `7.x` but instead of requiring the version to be set in two places (pipeline and `global.json`) it would allow it to be controlled from a single place and always consistent with the developer's environment.

**Comment by vsilvar** [comment]
Indeed, I omitted the fact that VS/CLI doesn't actually install the SDK for simplicity.
And while I agree that the only way of making sure the developer's environment and the build server are completely consistent is to specify a specific SDK version, when a developer specifies a `rollForward` policy, they are accepting a degree of variability between other developer's machines and possibly the build server.
But they are most likely doing so with the intent that the newest version would be used if available (and when possible, in the installation scenario?).

So, while I do understand the current behavior and don't even see it as a bug, I think we agree that my suggested behavior would be a nice improvement, and thus why I requested it as a feature.

**Comment by stijnherreman** [comment]
The task already has some support for a roll-forward policy in `versionfetcher.ts` by using `x` as a version part, but that is not supported in `global.json`. It certainly seems doable to implement full support for `global.json` while also keeping the existing task behaviour intact.

**Comment by Porges** [comment]
I do consider the current behaviour a bug because it means that anyone using something like `latestFeature` will not receive any security updates.

**Comment by char-iot** [comment]
up vote

**Comment by dariuszparys** [comment]
Please also include `latestPatch`. For instance

```json
{
    "sdk": {
        "version": "8.0.100",
        "rollForward": "latestPatch"
    }
}
```

**Comment by Barsonax** [comment]
Also running into this, now we are missing security updates due to this bug.

This should really get fixed soon as I suspect this is affecting alot of ppl who might not even realize it till its too late.

**Comment by EnableRyan** [comment]
Very surprised to find this out as well - I assumed "useGlobalJson" would use *all* of global.json, not just sdk->version. Will be keeping an eye on this for sure.

**Comment by matt1munich** [comment]
Very frustrating.  Following to see what comes of this issue.

**Comment by doubleyewdee** [comment]
In addition to the mystifying versioning scheme of .NET (using a bespoke "channels" concept in the patch version instead of a minor version is, bluntly, user-hostile) this contributed to me burning quite a lot of time trying to solve mysterious local build breaks which weren't happening in CI. Somehow, between .NET 8.0.3xx and .NET 8.0.4xx, new style enforcement got turned on. My developer machines (multiple, macOS + WSL), all hit style errors that broke the build on my machine, but never broke CI. Because CI did not honor what my `global.json` explicitly says, which is to roll forward SDK to the latest feature version, and so there was a (from a "user of many versioned things" perspective) relatively minor version change at the heart of this strange behavior that was *very* easy to miss.

I could, I suppose, lock on `rollForward` in my team's repository, but this feels like a poor solution vs. the task itself honoring the `global.json` behavior correctly. What I *want* is the `rollForward` behavior as checked-in, but consistency of behavior apparently suffers from this and I am left hunting for unsatisfactory workarounds.

**Comment by romansp** [comment]
Disappointed to find that `rollForward` in global.json isn’t supported. Kind of defeats the whole purpose of `useGlobalJson: true` to be honest. Any chance support for it is on the roadmap?
